frontend/src/components/LoginPage.tsx

```diff
@@ -1,32 +1,30 @@
 import {FormEvent, useContext, useState} from "react";
 import {AuthContext} from "../context/AuthProvider";
 import "../style/LoginPage.css"
+import useUsername from "../hooks/useUsername";
 
-type LoginPageProps = {
-    username: string;
-    setUsername:  (username:string) => void;
-}
-
-export  default function LoginPage({username, setUsername}: LoginPageProps) {
+export default function LoginPage() {
     const [password, setPassword] = useState<string>("");
+    const {username, setUsername} = useUsername();
```

Comment:
It is here OK, however, isn't it dangerous to use hooks at several Pages and Components?